usersimcrs/items/ratings.py

```diff
@@ -151,21 +187,11 @@ def create_split(
         if historical_ratio > 1.0 or historical_ratio < 0.0:
             raise ValueError("historical_ratio is bounded in [0,1]")
 
-        # Determine the number of items to use as historical data.
-        nb_historical_items = int(historical_ratio * len(self._item_ratings))
-
         historical_ratings = Ratings(self._item_collection)
         ground_truth_ratings = Ratings(self._item_collection)
 
-        for i, (item_id, user_ratings) in enumerate(self._item_ratings.items()):
-            for user_id, rating in user_ratings.items():
-                if i < nb_historical_items:
-                    historical_ratings.add_user_item_rating(
-                        user_id, item_id, rating
-                    )
-                else:
-                    ground_truth_ratings.add_user_item_rating(
-                        user_id, item_id, rating
-                    )
+        sampler(
+            self, historical_ratio, historical_ratings, ground_truth_ratings
+        )
```

Comment:
The outer loop over the users should be here, and the sampler samples ratings for a given user.